src/compiler/crystal/semantic/call_error.cr

```diff
@@ -394,10 +394,12 @@ class Crystal::Call
       str << "\n - "
       append_def_full_name a_def.owner, a_def, arg_types, str
       if defs.size > 1 && a_def.same?(matched_def)
-        str << colorize(" (trying this one)").blue
+        str << " (trying this one)".colorize.blue
+        str << "\e[1m" # Make the rest of the string bold
```

Comment:
This seems hackish... isn't there any other way to avoid using raw ansi escape codes?

ditto below

Reply:
Yes, I know. I would really like to avoid this but I couldn't find a way so far.
Let's cut the problem down to this:
```cr
require "colorize"

str = IO::Memory.new

str << "\n - "
str << "item 1"
str << "\n - "
str << "item 2"

str << " (did you mean this one?)".colorize.yellow

str << "\n - "
str << "item 3"

puts str.to_s.colorize.bold
```

![did you mean](https://user-images.githubusercontent.com/35064754/50680620-a6755180-1008-11e9-9961-fd931a16c32a.png)

Internally the string is:
```cr
"\e[1m\n - item 1\n - item 2\e[33m (did you mean this one?)\e[0m\n - item 3\e[0m"
 ^                          ^                              ^               ^
 Bold                       Yellow                         Terminates      Terminates
                                                           Yellow          Bold
```
The problem here is that the `Terminates Yellow` escape sequence doesn't just terminate `Yellow` but also `Bold` while actually the *last* sequence (`Terminates Bold`) is supposed to terminate `Bold`.
And that's fixed by appending a new `Bold` (`"\e[1m"`) after `Terminates Yellow`.

![did you mean](https://user-images.githubusercontent.com/35064754/50683539-ada15d00-1012-11e9-8900-176a973a42e7.png)

Reply:
You just need to call `colorize.bold` on each individual component instead of the entire list.

Reply:
Might seem a bit irrational as it could normally be summarized. But that's the way these colorize  commands work. You can't nest them.

Reply:
I figured out now why the specs are failing.
Currently I'm colorizing each item invidually bold but the whole error message is also getting colorized bold at the end nearly before it's getting raised. Thus the error ends in `\e[0m\e[0m`. So I could fix the specs by manually appending `\e[0m` at the end. Or by colorizing everything here in this string build invidually too: [`call_error.cr:553`](https://github.com/crystal-lang/crystal/blob/36de8d89131aad4bda97a83221898fa07f770d1a/src/compiler/crystal/semantic/call_error.cr#L553) and then disabling that the whole string gets colorized bold by adding an ugly parameter here: [`exception.cr:140`](https://github.com/crystal-lang/crystal/blob/9c42d31daf1331e31ba2c07679571acea699fa7a/src/compiler/crystal/semantic/exception.cr#L140) just for this one case. But I don't think that's the right way.
But I can't really think of anything else which does not involve using ANSI escape sequences. Maybe modifying `Colorize` but I also don't think that's the right way because the user just expects `\e[0m\e[0m` to be at the end in this case.

Reply:
@r00ster91 if you want to reset the colors, you can use `Colorize.reset` instead of directly the raw ANSI escape code.

Reply:
But `Colorize.reset` only works for `IO`'s, not strings.